packages/form-layout/src/vaadin-form-layout-mixin.js

```diff
@@ -528,6 +528,7 @@ export const FormLayoutMixin = (superClass) =>
     __maxColumnsChanged(maxColumns, autoResponsive) {
       if (autoResponsive) {
         this.style.setProperty('--_max-columns', maxColumns);
+        this.__updateCSSGridLayout();
```

Comment:
Do we need to keep `setProperty` here considering that it will anyway be updated by _updateLayout?

```suggestion
        this._updateLayout();
```

Reply:
It turns out if I move the rendered column count calculation to the bottom of the `__updateCSSGridLayout()` method, then I can remove the `setProperty` call here. I was facing some issues of the rendered column count being calculated before the max columns are set. Done.